game-core/src/main/java/games/strategy/engine/data/GameMap.java

```diff
@@ -127,19 +137,41 @@ public Territory getTerritory(final String s) {
   /**
    * Returns all neighbors within a certain distance of the starting territory that match the
    * condition. Does NOT include the original/starting territory in the returned Set.
+   *
+   * @param territory starting territory
+   * @param distance distance to check
+   * @param cond the neighbor territory must match this condition
    */
```

Comment:
Side-note, the getNeighbors code is largely only used by AI. It's something for us to consider as every logical graph operation involving the map does not scale if it's all found within `GameMap`. My thoughts are a `Territory` object having a collection of neighbors would help avoid the need to get all data from `GameMap`, and then also the AI code could have the path finding algorithms localized in the AI packages.